cmd/pipecd/ops.go

```diff
@@ -79,6 +80,31 @@ func (s *ops) run(ctx context.Context, t cli.Telemetry) error {
 		return err
 	}
 
+	// Prepare sql database.
+	if cfg.Datastore.Type == model.DataStoreMySQL {
+		mysqlEnsurer := mysqlensurer.NewMySQLEnsurer(
+			cfg.Datastore.MySQLConfig.URL,
+			cfg.Datastore.MySQLConfig.Database,
+			cfg.Datastore.MySQLConfig.UsernameFile,
+			cfg.Datastore.MySQLConfig.PasswordFile,
+			t.Logger,
+		)
+		err := mysqlEnsurer.EnsureSchema(ctx)
+		if err != nil {
+			t.Logger.Error("failed to prepare sql database", zap.Error(err))
+			return err
+		}
+		// No need to run this create indexes operation in routine because it runs asynchronously.
+		// ref: https://dev.mysql.com/doc/refman/8.0/en/innodb-online-ddl-operations.html#online-ddl-index-operations
+		err = mysqlEnsurer.EnsureIndexes(ctx)
+		if err != nil {
+			t.Logger.Error("failed to create required indexes on sql database", zap.Error(err))
+			return err
+		}
+		// Close connection held by the client.
+		mysqlEnsurer.Close()
```

Comment:
nit: wrap those lines into a function and use defer to ensure closing.

Reply:
Nice suggestion 🙏 addressed on https://github.com/pipe-cd/pipe/pull/1699/commits/46d79b156a65cf5248b98a5b9bdf986534aa94bf